API: named tags broken when uploading data to a library

**Describe the bug**
Some change over the last week broke the BioBlend test `TestGalaxyLibraries.test_upload_file_contents_with_tags` with the following error:

```
    @test_util.skip_unless_galaxy("release_19.09")
    def test_upload_file_contents_with_tags(self):
        datasets = self.gi.libraries.upload_file_contents(self.library["id"], FOO_DATA, tags=["name:foobar", "barfoo"])
        dataset_show = self.gi.libraries.show_dataset(self.library["id"], datasets[0]["id"])
>       assert listify(dataset_show["tags"]) == ["name:foobar", "barfoo"]
E       AssertionError: assert ['name:name:f...'name:barfoo'] == ['name:foobar', 'barfoo']
E         
E         At index 0 diff: 'name:name:foobar' != 'name:foobar'
E         
E         Full diff:
E           [
E         -     'name:foobar',
E         +     'name:name:foobar',...
```

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: dev
Commit: dev as 2024-08-13


**To Reproduce**
Steps to reproduce the behavior:
1. See https://github.com/galaxyproject/bioblend/actions/runs/10361199784/job/28681057831

**Expected behavior**
BioBlend test not failing on Galaxy dev.

--- a/lib/galaxy/tools/actions/upload_common.py
+++ b/lib/galaxy/tools/actions/upload_common.py
@@ -195,8 +195,9 @@ def __new_library_upload(trans, cntrller, uploaded_dataset, library_bunch, tag_h
         tag_handler.apply_item_tag(item=ldda, user=trans.user, name="name", value=tag_from_filename, flush=False)
 
     if tags_list := uploaded_dataset.get("tags", False):
-        for tag in tags_list:
-            tag_handler.apply_item_tag(item=ldda, user=trans.user, name="name", value=tag, flush=False)
+        new_tags = tag_handler.parse_tags_list(tags_list)
+        for tag in new_tags:
+            tag_handler.apply_item_tag(item=ldda, user=trans.user, name=tag[0], value=tag[1], flush=False)
 
     trans.sa_session.add(ldda)
     if state:

--- a/lib/galaxy_test/api/test_tags.py
+++ b/lib/galaxy_test/api/test_tags.py
@@ -112,6 +112,11 @@ class TestLibraryDatasetTags(TagsApiTests):
         item = item_response.json()
         return item
 
+    def test_upload_file_contents_with_tags(self):
+        initial_tags = ["name:foobar", "barfoo"]
+        ld = self.library_populator.new_library_dataset(name=f"test-library-dataset-{uuid4()}", tags=initial_tags)
+        assert ld["tags"] == initial_tags
+
 
 class TestPageTags(TagsApiTests):
     api_name = "pages"

--- a/lib/galaxy_test/base/populators.py
+++ b/lib/galaxy_test/base/populators.py
@@ -2687,6 +2687,7 @@ class LibraryPopulator:
             "upload_option": upload_option,
             "file_type": kwds.get("file_type", "auto"),
             "db_key": kwds.get("db_key", "?"),
+            "tags": kwds.get("tags", []),
         }
         if kwds.get("link_data"):
             create_data["link_data_only"] = "link_to_files"

